crates/sui-graphql-rpc/src/server/data_provider.rs

```diff
@@ -216,6 +221,108 @@ fn convert_bal(b: sui_json_rpc_types::Balance) -> Balance {
         total_balance: BigInt::from_str(&format!("{}", b.total_balance)).unwrap(),
     }
 }
+pub(crate) async fn fetch_gas_input(cl: &SuiClient, digest: &String) -> Result<GasInput> {
+    let tx_block = cl
+        .read_api()
+        .get_transaction_with_options(
+            TransactionDigest::from_str(digest)?,
+            SuiTransactionBlockResponseOptions::full_content(),
+        )
+        .await?;
+    let tx = tx_block.transaction.unwrap().data;
+    let gas_data = tx.gas_data();
+
+    let payment_obj_ids = gas_data
+        .payment
+        .clone()
```

Comment:
+1, clone should not be necessary here:

```rust
let payment_obj_ids: Vec<_> = gas_data
    .payment
    .iter()
    .map(|o| o.object_id)
    .collect();
```

Also, small nit: If you're collecting into a variable, it can be neater to put the type ascription on the variable declaration rather than the explicit type parameter to `collect`.